Zend/tests/return_types/never_disallowed3.phpt

```diff
@@ -0,0 +1,19 @@
+--TEST--
+never return type: unacceptable cases: implicit return
+--FILE--
+<?php
+
+function foo(): never {
+    if (false) {
+        throw new Exception('bad');
+    }
+}
+
+foo();
+?>
+--EXPECTF--
+Fatal error: Uncaught TypeError: foo(): Nothing was expected to be returned in %s:%d
+Stack trace:
+#0 %s(%d): foo()
+#1 {main}
+  thrown in %s on line %d
```

Comment:
this is a bit cryptic, and i can see how people might be confused by it.

`foo()` never returns explicitly, but `null` is returned implicitly, should the exception message mention that?

Reply:
Now the message reads "never-returning function must not implicitly return"